crates/net/eth-wire/src/types/broadcast.rs

```diff
@@ -204,10 +207,80 @@ pub struct NewPooledTransactionHashes68 {
     pub hashes: Vec<H256>,
 }
 
+impl Encodable for NewPooledTransactionHashes68 {
+    fn length(&self) -> usize {
+        #[derive(RlpEncodable)]
+        struct EncodableNewPooledTransactionHashes68<'a> {
+            types: Bytes,
+            sizes: &'a Vec<usize>,
+            hashes: &'a Vec<H256>,
+        }
+
+        let encodable = EncodableNewPooledTransactionHashes68 {
+            types: Bytes::from(self.types.clone()),
+            sizes: &self.sizes,
+            hashes: &self.hashes,
+        };
+
+        encodable.length()
+    }
+    fn encode(&self, out: &mut dyn bytes::BufMut) {
+        #[derive(RlpEncodable)]
+        struct EncodableNewPooledTransactionHashes68<'a> {
+            types: Bytes,
+            sizes: &'a Vec<usize>,
+            hashes: &'a Vec<H256>,
+        }
+
+        let encodable = EncodableNewPooledTransactionHashes68 {
+            types: Bytes::from(self.types.clone()),
```

Comment:
oh I didn't realize that we can't borrow under these circumstances.

this is pretty wasteful because now we need to clone a vec for every message we send...

So for perf reasons, we should change it to `NewPooledTransactionHashes68 {types: Bytes}` directly.

Reply:
or mirroring the  `Bytes::encode` impl here would also be possible

Reply:
ended up mirroring the `Bytes::encode` impl, in that `EncodableNewPooledTransactionHashes68` now uses `&[u8]` instead of `Bytes`, so we do not need to clone